src/ResourceManagement/Network/NicIPConfigurationImpl.cs

```diff
@@ -415,5 +409,42 @@ NetworkInterface.Update.IUpdate ISettable<NetworkInterface.Update.IUpdate>.Paren
         {
             return Parent;
         }
+
+        ///GENMHASH:8CC0232B83175AB854D257B255BB0AF5:66D59B6CE3C9166272F6266E8BFA6647
+        public NicIPConfigurationImpl WithoutApplicationGatewayBackends()
+        {
+            Inner.ApplicationGatewayBackendAddressPools = null;
+            return this;
+        }
+
+        ///GENMHASH:CB5C1E0F5FDBFFBE97A5C977559D67FF:2D595A527D306417A72C1C52A5906DB3
+        private IList<BackendAddressPoolInner> EnsureLoadBalancerBackendAddressPools()
+        {
+            Inner.LoadBalancerBackendAddressPools = Inner.LoadBalancerBackendAddressPools ?? new List<BackendAddressPoolInner>();
+            return Inner.LoadBalancerBackendAddressPools;
+        }
+
+        ///GENMHASH:B9A8F99DA807CD48CCF5ADBB6DB5C1C4:6D152F32C433E4139EE4F5562A275A64
+        public NicIPConfigurationImpl WithExistingApplicationGatewayBackend(IApplicationGateway appGateway, string backendName)
+        {
+            if (appGateway != null)
+            {
+                var pool = appGateway.Inner.BackendAddressPools.FirstOrDefault(o => o.Name.Equals(backendName, StringComparison.OrdinalIgnoreCase));
+                if (pool != null)
+                {
+                    EnsureAppGatewayBackendAddressPools().Add(pool);
+                    return this;
+                }
+            }
+            return null;
```

Comment:
>return null; [](start = 12, length = 12)

Please double-check  returning null in the fluent flow.

Reply:
yes, this is intentional - same rationale as explained earlier

Reply:
please throw instead of returning null.

Reply:
ok